src/inventory_ui.cpp

```diff
@@ -910,7 +915,21 @@ item_location game::inv_map_splice(
                     }
                 }
             }
-            inv_s.make_item_list( slices.back(), pos == g->u.pos() ? &ground_cat : &nearby_cat );
+            if ( match_items_found ) {
+                if( pos == g->u.pos() ) {
+                    inv_s.make_item_list( slices.back(), &ground_cat );
+                } else {
+                    int dist = square_dist( g->u.pos(), pos );
+                    std::string dir_name = direction_name_short( direction_from( g->u.pos(), pos ) );
+                    std::string cat_id = string_format( "NEARBY %d", nearby_rank );
+                    std::string cat_name = string_format( _( "NEARBY %d %s:" ), dist, trim( dir_name ).c_str() );
+
+                    std::unique_ptr<item_category> nearby_cat( new item_category( cat_id, cat_name, nearby_rank ) );
+                    nearby_rank -= 10;
+                    inv_s.make_item_list( slices.back(), nearby_cat.get() );
+                    nearby_cats.push_back( std::move( nearby_cat ) );
```

Comment:
The move is required, `nearby_cat` is a `unique_ptr`, it can not be copied (which is what `push_back` would do without the move), but it can be moved into the vector. Alternatively, one could use a shared pointer, it can be copied.